gui/default/syncthing/core/syncthingController.js

```diff
@@ -1402,6 +1402,11 @@ angular.module('syncthing.core')
                 autoNormalize: true
             };
             $scope.editingExisting = false;
+            $scope.newIgnores = {
+                edited: false,
```

Comment:
> Well at the point you press save things are set it stone. You could easily add folder paused, set ignores and unpause the folder as part of the addition process

That is exactly what I do in this PR. It means that at the moment when the user is editing ignores, the existing ignores cannot be loaded. At your request when saving the ignores a preexisting ".stignore" is overwritten, not appended to.

> Also, I feel the end state would either be load stuff for unadded folders and allow modifying (but that's a security hole), or overwriting

I half-heartedly proposed such an extension to the ignores API point but that was not met by acclaim. As you and calmh pointed out and I completely agree, if someone puts a ".stignore" in place he wont just overwrite by accident, so I think just overwriting an existing file as it is currently done is simplest and safest.